solo/cli/__init__.py

```diff
@@ -107,7 +108,10 @@ def ls():
     print(":: Solos")
     for c in solos:
         descriptor = c.dev.descriptor
-        print(f"{descriptor['path']}: {descriptor['product_string']}")
+        if "serial_number" in descriptor:
+            print(f"{descriptor['serial_number']}: {descriptor['product_string']}")
+        else:
+            print(f"{descriptor['path']}: {descriptor['product_string']}")
```

Comment:
You can make the above be:

    print(f"{descriptor.get('serial_number', descriptor['path'])}: {desciptor['product_string']}")

if that's a little clearer.

Reply:
I think that's less clear.

...says someone who keeps bumping
https://github.com/solokeys/solo-python/blob/6915722f9bd8a25f3eab982cbcc1544666bc10e5/.flake8#L8-L9
to account for `solo key update` complexity :grin: